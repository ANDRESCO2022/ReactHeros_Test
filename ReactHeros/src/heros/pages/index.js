export * from './DcPages';
export * from './MarbelPages';
export * from './HeroPages';
export * from './SearchPages';
