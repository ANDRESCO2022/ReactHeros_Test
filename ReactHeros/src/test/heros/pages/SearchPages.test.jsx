import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPages } from '../../../heros';

const mockedUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('pruebas  en <SearchPages/>', () => {
  beforeEach(() => jest.clearAllMocks());
  test('debe de mostrarse correctamente  con los  valores por  defecto', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPages />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test('debe de mostrar  a batman y el query con el queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPages />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');
    expect(input.value).toBe('batman');
    const img = screen.getByRole('img');
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg');
    const alertDanger = screen.getByLabelText('alert-danger');
    expect(alertDanger.style.display).toBe('none');
  });
  test('debe de  mostrar un error  si no se  encuentra el hero', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123']}>
        <SearchPages />
      </MemoryRouter>
    );
    const alertDanger = screen.getByLabelText('alert-danger');
    expect(alertDanger.style.display).toBe('');
  });
  test('debe  de llamar el navigate a la  pantalla nueva', () => {
    const inputValue = 'superman';
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPages />
      </MemoryRouter>
    );
    const input = screen.getByRole('textbox');
    fireEvent.change(input, {
      target: { name: 'searchText', value: inputValue },
    });
    const form = screen.getByRole('form');
    fireEvent.submit(form);
    expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
  });
});
