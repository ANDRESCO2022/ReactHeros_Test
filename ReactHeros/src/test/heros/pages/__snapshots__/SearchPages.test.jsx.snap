// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas  en <SearchPages/> debe de mostrarse correctamente  con los  valores por  defecto 1`] = `
<div>
  <h1>
    search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <form
        aria-label="form_label"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder=" Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <div
        class="alert alert-primary  animate__animated animate__fadeIn"
      >
        Shearh a hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        not exist hero whith 
        <b />
      </div>
    </div>
  </div>
</div>
`;
